### DISCUSSION TITLE
Can you please integrate EEP A5-13-01 (Eltako MS) Weatherstation? [DONE]

### DISCUSSION THREAD

**Discussion by Ekgev** [mak-gitdev/HA_enoceanmqtt#24] [Request] [answered]
Hi, that would be really helpful. Please let me know whether I can help with logfiles or so. Thanks

**Comment by mak-gitdev** [comment, marked as answer]
Hi,

A5-13-01 has been added with the latest dev release.
Please report if everything is OK. Thanks

**Reply by mak-gitdev** [reply]
https://community.home-assistant.io/t/enocean-mqtt-weather-station/512291/5

**Reply by chma73** [reply]
Hi, I also want to use Eltako MS, I added it in HA, but I don't get any results from the station. The sensors are listed but the result is unknown. Can you confirm that the integration works?

**Reply by mak-gitdev** [reply]
Hi @chma73,

The result may be unknown because your device didn't generate data yet.
Do your sensors still have unknown values ?

If so, which version are you using ?
Could you share you device file as well as the log file ?

**Reply by chma73** [reply]
Hi @mak-gitdev,
- the sensor should generate data, because it's now 24 hours since I implement the sensor
- yes it's still unknown
- which version do you mean? EnOcean MQTT (dev) Current version: 0.1.26 
- do you mean this; where can I find the device log file or do you mean the general log file?
[Dach_Wetterstation]
address      = 0xFFA46E80
rorg            = 0xA5
func            = 0x13
type            = 0x01

**Reply by schlegel** [reply]
@chma73 I can confirm that it works with Eltako MS. As a side note: In the first place my Eltako MS was broken and I didn't get any information from the device. I was able to check it with a second home automation solution which was installed already in my house (wibutler). Do you have an alternative way to check the Eltako MS? It had to be replaced in my case.

**Reply by chma73** [reply]
Yes I checked it with homee and with homee the Eltako MS works

**Reply by mak-gitdev** [reply]
@chma73,

You use the correct version so OK for that.

Regarding the log file, it should be `/config/enoceanmqtt.log` directory in HA. Note that `/config` is the directory where you have your `configuration.yaml` file.

Also please, what is your device reference ?

**Reply by chma73** [reply]
hi @mak-gitdev, hope the upload works
[enoceanmqtt.log](https://github.com/mak-gitdev/HA_enoceanmqtt/files/12287760/enoceanmqtt.log)
what do you mean with device reference - the Eltako MS runs actual at homee system without any problem

**Reply by mak-gitdev** [reply]
@chma73,

> what do you mean with device reference

Sorry, I missed you said your device was Eltako MS :sweat_smile: 

I read your log and there are a lot of unknown devices. Are you sure the address of your MS is 0xFFA46E80 ?
To check that, generate a learn request from your MS and have a look at the log.
If you have a new entry with "unknown device", then your address the one indicated alongside the "unknown device" message.

**Reply by chma73** [reply]
at the Eltako device I found the ID see picture
![Ohne Titel](https://github.com/mak-gitdev/HA_enoceanmqtt/assets/137674516/d900b97d-5126-46ae-bd6e-b29d46c9828f)

how can I generate a learn request?
 
I know that I have still unknown devices this are Plugs from Permundo and Windowsensors from Eimsig - but I don't know how I can find the right rorg, func and type

**Reply by chma73** [reply]
aha I found the issue I need to use the last 8 numbers from the ID

**Reply by mak-gitdev** [reply]
@chma73,

So for your MS, you should use 0x050848DD.

For your other devices, it should be indicated in their user manual.

Give me the references of your devices, I can help you find the correct EEP.

**Reply by chma73** [reply]
@mak-gitdev thanks for the hint I just released it :-)

the other known devices are Permundo Smart Plug PSC234; Permundo Smart Plug PSC234; [EM-FGSE-01] EiMSIG Fenster+Glasbruchsensor EnOcean; AFRISO WaterSensor con

**Reply by mak-gitdev** [reply]
@chma73 

>AFRISO Watersensor con

From this [link](https://www.afriso.com/en/PM/Domestic-technology/AFRISO-Smart-Home/Conductivity-water-sensor-WaterSensor-con), the EEP is A5-30-03. This is not yet supported.

> Permundo Smart Plug PSC234

It seems to be D2-01-09. This EEP is not yet supported but you can use D2-01-0E for the moment.

> [EM-FGSE-01] EiMSIG Fenster+Glasbruchsensor EnOcean

It seems to be A5-14-xx. You can try A5-14-01 or A5-14-0A.
For these EEP, there is a bug in the mapping file. Please have a look at #72.

**Reply by chma73** [reply]
@mak-gitdev thanks for your support, I could solve some unknown device; but I don't want to high jack this chat; is there any chance that A5-30-3 and D2-01-09 will be implemented?

**Reply by mak-gitdev** [reply]
@chma73,

Yes those 2 are not difficult.
Simply open a new discussion with the EEPs you want me to add in the title, so that I can track your request.
